content/information-distribution/signature.md

```diff
@@ -17,28 +17,46 @@ Informo signed events are built using a common structure, inspired from [Matrix
-| `signature`  | `string` | Generated signature                                                               |
-| `signed`     | `object` | Content that has been signed                                                      |
-
-<!-- 🔧: Need to do some research on Megolm and Matrix APIs around encryption and key
-management -->
+|    Field     |      Type      |                                                                 Description                                                                 |
+|--------------|----------------|---------------------------------------------------------------------------------------------------------------------------------------------|
+| `signature`  | `string`       | Signature generated from `signed` using the private key from the same pair as the public key announced in `signed`'s `signatory_key` field. |
+| `signed`     | `SignedObject` | Object to be signed.                                                                                                                        |
+
+Where:
+
+* `SignedObject` is a map using the following structure:
+
+| Parameter       |       Type       | Req. |                                                  Description                                                  |
+|:----------------|:-----------------|:----:|:--------------------------------------------------------------------------------------------------------------|
+| `sender`        | `string`         |  x   | User who sent the event. **Must** match the Matrix user ID of who sent the event.                             |
+| `room_id`       | `string`         |  x   | Associated room ID. **Must** match the room ID where the event has been sent.                                 |
+| `type`          | `string`         |  x   | Type of the Matrix event.                                                                                     |
+| `state_key`     | `string`         |  !   | State key of the event. Required if the event is a state event.                                               |
+| `signatory`     | `string`         |      | The Matrix user ID who signed this data. If not present, defaults to the value defined in the `sender` field. |
+| `signatory_key` | `string`         |  x   | The public part of the key pair used for signing the event.                                                   |
+| `algorithm`     | `string`         |  x   | Algorithm used for signing the content.                                                                       |
+| `content`       | `object`         |  x   | The content of the event to be signed.                                                                        |
+
+If the value of the `sender` field does not match the user ID sending the signed
+Matrix event, or if the `room_id` does not match the room where the event is
+sent, then the signature **must** be considered as invalid.
```

Comment:
Maybe make this a bit more generic like:

> If the value of any field referring to a property with the same name for the signed event doesn't match the value for the said property, then the signature **must** be considered as invalid.

Since the list of properties that need to match with the event that's being signed is getting quite long.